signal.c

```diff
@@ -1561,3 +1566,13 @@ Init_signal(void)
 
     rb_enable_interrupt();
 }
+
+ #if defined(HAVE_WORKING_FORK)
```

Comment:
This condition doesn’t look matching the above.
And the function is always needed, so the condition can be in the function body.